Fix UTs for qnx7. part 2

BSolonenko opened the pull request:
This PR is **ready** for review.  
This PR includes implementation for  proposal [Porting SDL to QNX700 and Automotive Grade Linux (AGL) x64 platforms](https://github.com/LuxoftAKutsan/sdl_evolution/blob/port_sdl_on_qnx700_and_agl/proposals/nnnn-sdl_on_qnx700_agl.md)

### Summary
Unit tests have been fixed for: 

- application_manager_test
- telemetry_monitor_test
- security_manager_test

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

Commit:
The unit test of the application_manager_test has been fixed.

Commit:
The unit test of the telemetry_monitor_test has been fixed.

BSolonenko commented:
@AByzhynar, @mkorniichuk, @LuxoftAKutsan and @SNiukalov please review.

mkorniichuk commented on src/components/security_manager/test/CMakeLists.txt:
```diff
@@ -59,8 +59,14 @@ add_custom_target(generate_certificates
 )
 
+add_custom_target(copy_certificates
+  COMMAND /bin/bash -c "cp -r client server ${TEST_BIN_DIR}"
```
@BSolonenko I suggest you using ${CMAKE_COMMAND} -E copy ... instead and use **add_custom_command** with OUTPUT parameter, since the target is always outdated

BSolonenko replied:
@mkorniichuk this solution does not work for folders

LuxoftAKutsan commented on src/components/include/utils/byte_order.h:
```diff
@@ -35,12 +35,20 @@
+The ENDIAN_SWAP32() macro endian-swaps the value pointed to by num
+in place. Therefore, copying is necessary so as not to damage the
+original data.
+*/
```
Please delver it as a separate commit

BSolonenko replied:
@LuxoftAKutsan  Done in https://github.com/Ford-Luxoft/sdl_core/pull/30/commits/e8a0ccf961329f78b8528fcafb9172f71a013234

LuxoftAKutsan commented on src/components/telemetry_monitor/src/metric_wrapper.cc:
```diff
@@ -68,6 +68,10 @@ void MetricWrapper::Clear() {
   }
 }
 
```
Method can be const an return const value or even const reference

BSolonenko replied:
@LuxoftAKutsan fixed in https://github.com/Ford-Luxoft/sdl_core/pull/30/commits/3f042c33cb3c723382c0e0e040bc84a0158b1e70

Commit:
Source data corruption using BYTE_SWAP_32 has been fixed.

The ENDIAN_SWAP32() macro endian-swaps the value pointed to by num
in place. Therefore, copying is necessary so as not to damage the
original data.

Commit:
The unit test of the security_manager_test has been fixed.

mkorniichuk commented on src/components/utils/test/qdb_wrapper/start_test_qdb_server.sh:
```diff
@@ -11,10 +12,14 @@ fi
    mkdir -p $DB_BACKUP_PATH
 fi
-# Creating policy for the resumption
+# Creating policy for the policy
```
@BSolonenko policy for the policy?

BSolonenko replied:
@mkorniichuk  fixed in https://github.com/Ford-Luxoft/sdl_core/pull/30/commits/d2cf4a58dff4be7458458a5b23d94e0c0ecc9d0f

LuxoftAKutsan reviewed (approved):
Remove redundant #include <iostream>

AByzhynar commented on src/components/telemetry_monitor/include/telemetry_monitor/metric_wrapper.h:
```diff
@@ -51,7 +51,7 @@ class MetricWrapper {
   virtual Json::Value GetJsonMetric();
   virtual ~MetricWrapper();
-
+  virtual const utils::ResourseUsage& getResources();
```
@BSolonenko 
- named wrongly according to SDL Coding Style

BSolonenko replied:
@AByzhynar fixed in https://github.com/Ford-Luxoft/sdl_core/pull/30/commits/5e15fe123252c49dde19d2d32cd757f1ac7eacf8

AByzhynar commented on src/components/telemetry_monitor/src/metric_wrapper.cc:
```diff
@@ -68,6 +68,10 @@ void MetricWrapper::Clear() {
 }
 
+const utils::ResourseUsage& MetricWrapper::getResources() {
+  return *resources;
```
@BSolonenko `getResources()` is a part of public interface: therefore it is potential core crash with `SEGMENATION_FAULT`  :boom:  :boom:  :boom:

BSolonenko replied:
@AByzhynar fixed in https://github.com/Ford-Luxoft/sdl_core/pull/30/commits/5e15fe123252c49dde19d2d32cd757f1ac7eacf8

AByzhynar commented on src/components/telemetry_monitor/test/application_manager_metric_test.cc:
```diff
@@ -85,8 +85,8 @@ TEST(ApplicationManagerMetricWrapper, GetJsonMetric) {
   ApplicationManagerMetricWrapper metric_test;
-  utils::ResourseUsage* resources = utils::Resources::getCurrentResourseUsage();
   EXPECT_TRUE(metric_test.grabResources());
+  const utils::ResourseUsage& resources = metric_test.getResources();
```
@BSolonenko Don't you have a  :bomb: here?

BSolonenko replied:
@AByzhynar fixed in https://github.com/Ford-Luxoft/sdl_core/pull/30/commits/5e15fe123252c49dde19d2d32cd757f1ac7eacf8
